yt_dlp/update.py

```diff
@@ -110,49 +116,71 @@ def _sha256_file(path):
 
 
 class Updater:
-    def __init__(self, ydl):
+    def __init__(self, ydl, channel_or_tag=None):
         self.ydl = ydl
 
+        channel_or_tag = channel_or_tag or CHANNEL
+        if '.' in channel_or_tag or channel_or_tag in ('stable', 'latest'):
```

Comment:
Currently this does not handle `-U`/`--update`, which will store `True` as the value of `opts.update_self`, which then gets passed as the `channel_or_tag` kwarg, causing a TypeError.

We could do this:
```diff
diff --git a/yt_dlp/options.py b/yt_dlp/options.py
index f558278ff..16c48e21b 100644
--- a/yt_dlp/options.py
+++ b/yt_dlp/options.py
@@ -36,7 +36,7 @@
     remove_end,
     write_string,
 )
-from .version import __version__
+from .version import CHANNEL, __version__
 
 
 def parseOpts(overrideArguments=None, ignore_config_files='if_override'):
@@ -323,10 +323,10 @@ def _alias_callback(option, opt_str, value, parser, opts, nargs):
         help='Print program version and exit')
     general.add_option(
         '-U', '--update',
-        action='store_true', dest='update_self',
+        action='store_const', dest='update_self', const=CHANNEL,
         help=format_field(
             is_non_updateable(), None, 'Check if updates are available. %s',
-            default='Update this program to the latest version'))
+            default=f'Update this program to the latest {CHANNEL} version'))
     general.add_option(
         '--no-update',
         action='store_false', dest='update_self',
```

Reply:
> +            default=f'Update this program to the latest {CHANNEL} version'))

Readme is generated from this. So best to leave it as before imo

Reply:
The new fix might need more testing, which can be done using `REPOSITORY = 'Grub4K/actions-test'`. I will do a regular release on there now so we can test all possibilities.